application fails do deploy to standalone instance or cluster, but works on server instance

[//]: # "Please remove these comments"
[//]: # "Provide a general summary of the issue in the Title above"


## Environment Details
* GlassFish Version (and build number): 7.0.6  (commit: 9e82903b2afcab9478003fb5adb7bd6a466e4982)
* JDK version:  zulu17.44.15-ca-jdk17.0.8
* OS: Centos7
* Database: Oracle

----------

## Problem Description
Application migrated from javax+jdk11 compatible to jakarta+jdk17. When deployed in default server instance of domain1, works fine.

When we tried to deploy same war into cluster (or standalone instance), deployment fails with following exception:

```
[2023-07-19T10:50:02.729035+02:00] [GF 7.0.6] [SEVERE] [AS-WEB-CORE-00108] [jakarta.enterprise.web.core] [tid: _ThreadID=1 _ThreadName=main] [levelValue: 1000] [[
  ContainerBase.addChild: start: 
org.apache.catalina.LifecycleException: org.glassfish.deployment.common.DeploymentException: Error in generating security policy for AxaAttendance-2.06 -- Error in generating security policy for AxaAttendance-2.06 -- Operation invoked on closed or deleted PolicyConfiguration.
	at org.apache.catalina.core.StandardContext.start(StandardContext.java:4467)
	at com.sun.enterprise.web.WebModule.start(WebModule.java:551)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:935)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:917)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:644)
	at com.sun.enterprise.web.WebContainer.loadWebModule(WebContainer.java:1795)
	at com.sun.enterprise.web.WebContainer.loadWebModule(WebContainer.java:1483)
	at com.sun.enterprise.web.WebApplication.start(WebApplication.java:88)
	at org.glassfish.internal.data.EngineRef.start(EngineRef.java:97)
	at org.glassfish.internal.data.ModuleInfo.start(ModuleInfo.java:262)
	at org.glassfish.internal.data.ApplicationInfo.start(ApplicationInfo.java:353)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:535)
	at com.sun.enterprise.v3.server.ApplicationLoaderService.processApplication(ApplicationLoaderService.java:406)
	at com.sun.enterprise.v3.server.ApplicationLoaderService.postConstruct(ApplicationLoaderService.java:242)
	at org.jvnet.hk2.internal.ClazzCreator.postConstructMe(ClazzCreator.java:356)
	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:410)
	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:466)
	at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.findOrCreate(AsyncRunLevelContext.java:281)
	at org.glassfish.hk2.runlevel.RunLevelContext.findOrCreate(RunLevelContext.java:65)
	at org.jvnet.hk2.internal.Utilities.createService(Utilities.java:2109)
	at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:93)
	at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:67)
	at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.oneJob(CurrentTaskFuture.java:1213)
	at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.run(CurrentTaskFuture.java:1144)
	at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpOneLevel.run(CurrentTaskFuture.java:762)
	at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpAllTheWay.go(CurrentTaskFuture.java:561)
	at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture.go(CurrentTaskFuture.java:133)
	at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture.go(CurrentTaskFuture.java:128)
	at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.proceedTo(AsyncRunLevelContext.java:511)
	at org.glassfish.hk2.runlevel.internal.RunLevelControllerImpl.proceedTo(RunLevelControllerImpl.java:48)
	at com.sun.enterprise.v3.server.AppServerStartup.proceedTo(AppServerStartup.java:498)
	at com.sun.enterprise.v3.server.AppServerStartup.run(AppServerStartup.java:302)
	at com.sun.enterprise.v3.server.AppServerStartup.doStart(AppServerStartup.java:209)
	at com.sun.enterprise.v3.server.AppServerStartup.start(AppServerStartup.java:200)
	at com.sun.enterprise.glassfish.bootstrap.GlassFishImpl.start(GlassFishImpl.java:55)
	at com.sun.enterprise.glassfish.bootstrap.GlassFishDecorator.start(GlassFishDecorator.java:38)
	at com.sun.enterprise.glassfish.bootstrap.osgi.EmbeddedOSGiGlassFishImpl.start(EmbeddedOSGiGlassFishImpl.java:50)
	at com.sun.enterprise.glassfish.bootstrap.GlassFishDecorator.start(GlassFishDecorator.java:38)
	at com.sun.enterprise.glassfish.bootstrap.osgi.OSGiGlassFishImpl.start(OSGiGlassFishImpl.java:46)
	at com.sun.enterprise.glassfish.bootstrap.GlassFishMain$Launcher.launch(GlassFishMain.java:144)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at com.sun.enterprise.glassfish.bootstrap.GlassFishMain.main(GlassFishMain.java:93)
	at com.sun.enterprise.glassfish.bootstrap.ASMain.main(ASMain.java:31)
Caused by: org.glassfish.deployment.common.DeploymentException: Error in generating security policy for AxaAttendance-2.06 -- Error in generating security policy for AxaAttendance-2.06 -- Operation invoked on closed or deleted PolicyConfiguration.
	at org.glassfish.exousia.modules.locked.SimplePolicyConfiguration.assertStateIsOpen(SimplePolicyConfiguration.java:686)
	at org.glassfish.exousia.modules.locked.SimplePolicyConfiguration.removeUncheckedPolicy(SimplePolicyConfiguration.java:425)
	at org.glassfish.exousia.AuthorizationService.removeStatementsFromPolicy(AuthorizationService.java:266)
	at com.sun.enterprise.security.web.integration.WebSecurityManager.release(WebSecurityManager.java:342)
	at com.sun.enterprise.security.ee.SecurityDeployer.loadWebPolicy(SecurityDeployer.java:238)
	at com.sun.enterprise.security.ee.SecurityDeployer.commitWebPolicy(SecurityDeployer.java:261)
	at com.sun.enterprise.security.ee.SecurityDeployer$AppDeployEventListener.event(SecurityDeployer.java:146)
	at org.glassfish.kernel.event.EventsImpl.send(EventsImpl.java:107)
	at com.sun.enterprise.web.WebContainer.afterServletContextInitializedEvent(WebContainer.java:569)
	at com.sun.enterprise.web.WebModule.contextListenerStart(WebModule.java:623)
	at org.apache.catalina.core.StandardContext.start(StandardContext.java:4450)
	... 45 more
]]
```
Full server log containing mentioned exception and following.
[server.log](https://github.com/eclipse-ee4j/glassfish/files/12105131/server.log)

## Steps to reproduce
There are no steps. 
If deployed to default server instance (_server-config_), exception does not happen.
If deployed to newly created cluster (or standalone) instance based on _default-config_.

Oracle JDBC driver present in lib folder and JDBC pool configured in domain.

Only way to reproduce would be to provide deployable war. 

## Impact of Issue
Application cannot be deployed to cluster.

We tried to delete _generated_ and _osgi-cache_ in domain and instance folders, didn't help.

--- a/appserver/pom.xml
+++ b/appserver/pom.xml
@@ -97,7 +97,7 @@
 
         <!-- Jakarta Security + Authentication/Authorization -->
         <soteria.version>3.0.3</soteria.version>
-        <exousia.version>2.1.0</exousia.version>
+        <exousia.version>2.1.1</exousia.version>
         <nimbus.version>9.31</nimbus.version>
         <jcip.version>1.0.2</jcip.version>
 

